Description: Avoid error messages when opening a new tomb
 Make sure control file exists and has user ownership before user writes to it.
Author: Sven Geuer <debmaint@g-e-u-e-r.de>
Bug: https://github.com/dyne/Tomb/issues/369
Last-Update: 2019-11-04
--- a/tomb
+++ b/tomb
@@ -2197,14 +2197,15 @@
 # {{{ Open
 
 _update_control_file() {
-	# replaces a control file with new contents and gives it user ownership
+	# make sure a control file exists, gives it user ownership
+	# and replaces it with new contents
 	# stdin = contents
 	# $1 = path to control file
 	# $2 = contents
 	[[ "$2" = "" ]] && return 1
 	_sudo touch "$1"
-	print "$2" > "$1"
 	_sudo chown ${_UID}:${_GID} "$1"
+	print "$2" > "$1"
 	_verbose "updated control file $1 = $2"
 }
 
